src/ngx_http_lua_req_body.c

```diff
@@ -312,6 +319,13 @@ ngx_http_lua_ngx_req_get_body_file(lua_State *L)
         return luaL_error(L, "request object not found");
     }
 
+/* http2 read body may break http2 stream process */
+#if (NGX_HTTP_V2)
+    if (r->main->stream) {
```

Comment:
Does this check are really needed?

Reply:
I have explain it in https://github.com/openresty/lua-nginx-module/issues/2172#issuecomment-1486378063